lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/ServiceAndContextConfigurationProvider.cs

```diff
@@ -0,0 +1,184 @@
+                    }
+                    else
+                    {
+                        _configurationProvider[evaluatorId] =
+                            GetMapTaskContextAndServiceConfiguration(_partitionDescriptors.Pop());
+                    }
+                }
+
+                _submittedEvaluators.Add(evaluatorId);
+                return _configurationProvider[evaluatorId];
+            }
+        }
+
+        private Tuple<IConfiguration,IConfiguration> GetMapTaskContextAndServiceConfiguration(IPartitionDescriptor partitionDescriptor)
+        {
+            IConfiguration contextConf = _groupCommDriver.GetContextConfiguration();
+            IConfiguration serviceConf = _groupCommDriver.GetServiceConfiguration();
+
+            var codecConfig =
+                TangFactory.GetTang()
+                    .NewConfigurationBuilder(
+                        new[]
+                        {
+                            StreamingCodecConfiguration<MapInputWithControlMessage<TMapInput>>.Conf.Set(
+                                StreamingCodecConfiguration<MapInputWithControlMessage<TMapInput>>.Codec,
+                                GenericType<MapInputWithControlMessageCodec<TMapInput>>.Class).Build(),
+                            StreamingCodecConfigurationMinusMessage<TMapOutput>.Conf.Build(),
+                            _configurationManager.MapInputCodecConfiguration
+                        }
+                    ).Build();
+
+            contextConf = Configurations.Merge(contextConf, partitionDescriptor.GetPartitionConfiguration());
+            serviceConf = Configurations.Merge(serviceConf, codecConfig,
+                _tcpPortProviderConfig);
+
+            return new Tuple<IConfiguration, IConfiguration>(serviceConf, contextConf);
+        }
+
+        private Tuple<IConfiguration, IConfiguration> GetUpdateTaskContextAndServiceConfiguration()
+        {
+            IConfiguration contextConf = _groupCommDriver.GetContextConfiguration();
+            IConfiguration serviceConf = _groupCommDriver.GetServiceConfiguration();
+
+            var codecConfig =
+                TangFactory.GetTang()
+                    .NewConfigurationBuilder(
+                        new[]
+                        {
+                            StreamingCodecConfiguration<MapInputWithControlMessage<TMapInput>>.Conf.Set(
+                                StreamingCodecConfiguration<MapInputWithControlMessage<TMapInput>>.Codec,
+                                GenericType<MapInputWithControlMessageCodec<TMapInput>>.Class).Build(),
+                            StreamingCodecConfigurationMinusMessage<TMapOutput>.Conf.Build(),
+                            _configurationManager.UpdateFunctionCodecsConfiguration
+                        }
+                    ).Build();
+
+            serviceConf = Configurations.Merge(serviceConf, codecConfig, _tcpPortProviderConfig);
```

Comment:
same as above: Call `_groupCommDriver.GetServiceConfiguration()` as part of the merge.